### DISCUSSION TITLE
Minimal C++ example needed

### DISCUSSION THREAD

**Discussion by jpswinski** [opengeospatial/geoparquet#164] [Ecosystem]
I'm a developer for an Earth science web service being developed by University of Washington and NASA/Goddard.  At the direction of one of our researchers, @scottyhq, we've added GeoParquet as an option for returning results to our end users.  This required building the GeoParquet file in C++ on our servers running in AWS, then streaming the file back to a client running Python that then opens the file as a GeoDataFrame using GeoPandas.

While I found the GeoParquet specification very easy to understand and follow, getting the code necessary to create a GeoParquet file in C++ ended up being very challenging.  I used Apache Arrow, and given what appears to be a good deal of churn in the Arrow/Parquet C++ library, the documentation I found was incomplete or outdated, and I could not find good examples for what I needed to do.

Having gone through the effort now, I've created a simple minimal example in C++ for writing a GeoParquet file that can then be read by GeoPandas.  It can be found at https://gist.github.com/jpswinski/13074fc773f92a529f98b274e5ad5283.  If this example program is helpful in any way to your documentation efforts, please feel free to use it however you see fit.

But if there are better ways to use the Apache Arrow library for writing GeoParquet files, then having a simple write-up or example program that shows how, would be extremely helpful.

**Comment by cholmes** [comment]
Thanks @jpswinski! And sorry for the super slow response. I'll put a link to this from geoparquet.org, to help other C++ developers.
